api/web3server.go

```diff
@@ -391,6 +395,21 @@ func (svr *web3Handler) call(in *gjson.Result) (interface{}, error) {
 	if err != nil {
 		return nil, err
 	}
+
+	result, _ := hex.DecodeString(ret)
+	if bytes.HasPrefix(result, []byte{0x08, 0xc3, 0x79, 0xa0}) {
+		strtyp, _ := abi.NewType("string", "", nil)
+		arguments := abi.Arguments{{Type: strtyp}}
+		args, err := arguments.Unpack(result[4:])
+		if err != nil {
+			log.Logger("api").Error("web3server",
+				zap.String("requestParams", "ret abi unpack"),
+				zap.Error(err))
+			return "0x" + ret, nil
+		}
+		return "0x" + ret, status.Error(codes.InvalidArgument, fmt.Sprintf("execution reverted: %s", args[0].(string)))
+	}
```

Comment:
we already have `ExecutionRevertMsg` field in `iotextypes.Receipt` at returned by `ReadContract`, please try to use it first if possible.

Reply:
Your code seems to attempt manual parsing of EVM return codes, which is not recommended. It may pose obstacles for future EVM upgrades. 
If `ExecutionRevertMsg` not work, maybe you can refer to this PR  in Ethereum, using `abi.UnpackRevert` for parsing purpose:
https://github.com/ethereum/go-ethereum/pull/21083/files#diff-c426ecd2f7d247753b9ea8c1cc003f21fa412661c1f967d203d4edf8163da344R912-R914